src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -1162,23 +1162,31 @@ private String[] SplitInternal(ReadOnlySpan<char> separators, int count, StringS
                 return new String[] { this };
             }
 
-            Span<int> sepList = Length < StackallocStringLengthLimit ? stackalloc int[Length] : new int[Length];
-            int numReplaces = MakeSeparatorList(separators, sepList);
+            Span<int> initialSpan = Length < StackallocStringLengthLimit ? stackalloc int[Length] : stackalloc int[StackallocStringLengthLimit];
```

Comment:
You can always allocate the full buffer here: `stackalloc int[StackallocStringLengthLimit]`.